core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -57,6 +57,7 @@
   private final HashMap<String,byte[]> cache;
   private final HashMap<String,ZcStat> statCache;
   private final HashMap<String,List<String>> childrenCache;
+  private final HashMap<String,Boolean> znodeExists = new HashMap<>();
```

Comment:
You appear to be using this map as a set (i.e. you never put(path, false)).  Why not simply make this a set?